### PR TITLE
feat: move create resource options to menu

### PR DESCRIPTION
**Pull request by karrui**
This pull request enhances the user interface of the Site Page by introducing a dropdown menu. The menu includes options to create folders, pages, and collections, improving the user experience. Refactoring involved importing new icons and components, and correcting a minor typo in variable naming.

### CODE REVIEW

**Comment by karrui** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/314?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#314** <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/314?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#313** <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/313?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#309** <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/309?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#308** <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/308?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#302** <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/302?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#301** <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/301?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @karrui and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/studio/src/pages/sites/[siteId]/index.tsx [hunk 1]
```diff
@@ -21,14 +31,23 @@ const SitePage: NextPageWithLayout = () => {
...
-              </Button>
-              <Button onClick={onPageCreateModalOpen} size="xs">
-                Create a new page
-              </Button>
-            </HStack>
+            <Menu isLazy size="sm">
+              <MenuButton as={Button} size="xs" justifySelf="flex-end">
+                Create new...
+              </MenuButton>
+              <MenuList>
```
Comment (karrui) [inline apps/studio/src/pages/sites/[siteId]/index.tsx:38]: i don't want to rebase anymore ><, but I've added a portal to the menulist (following [chakra docs](https://v2.chakra-ui.com/docs/components/menu#rendering-menu-in-a-portal)) so they are not constrained inside the table

Comment (karrui) [reply apps/studio/src/pages/sites/[siteId]/index.tsx:38]: nvm i still did it

**Review by seaerchin — APPROVED** [review]

**Comment by karrui** [comment]
### Merge activity

* **Jul 18, 11:17 PM EDT**: @karrui started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/opengovsg/isomer/314).

**Pull request merged**